Linking errors on Cori with PIO_VERSION=2, compiler=intel, mpilib=impi

When PIO_VERSION=2, compiler=intel, mpilib=impi, machine=cori-knl, the following linking errors can be reproduced with a simple X case:
```
Linking CXX executable spio_finfo.exe
...
/usr/bin/ld: /opt/cray/pe/netcdf-hdf5parallel/4.6.1.3/INTEL/16.0/lib/libnetcdf.a(libnetcdf4_la-nc4file.o): in function `att_read_var_callbk':
nc4file.c:(.text+0x71): undefined reference to `H5Aopen'
...
```
Not reproducible with PIO_VERSION=1 which does not build spio_finfo.exe
Not reproducible with PIO_VERSION=2, compiler=intel, mpilib=mpt

Seems to be related to dynamic linking of e3sm.exe
1) compiler=intel, mpilib=mpt, e3sm.exe is a static executable
2) For PIO_VERSION=2 (disable building of spio_finfo.exe), compiler=intel, mpilib=impi, e3sm.exe is a dynamic executable
3) For PIO_VERSION=1, compiler=intel, mpilib=impi, e3sm.exe is also a dynamic executable

Is it possible to build e3sm.exe as a static executable with mpilib=impi on Cori? Maybe that will also fix the linking errors of spio_finfo.exe.

@jayeshkrishna 
I am able to reproduce the link issue on Cori login node with stand-alone scorpio (use same modules as original E3SM X case):
```
source /opt/modules/default/init/sh
module rm craype craype-hugepages2M craype-mic-knl craype-haswell PrgEnv-intel PrgEnv-cray PrgEnv-gnu intel cce gcc cray-parallel-netcdf cray-hdf5-parallel pmi cray-mpich2 cray-mpich cray-netcdf cray-hdf5 cray-netcdf-hdf5parallel cray-libsci papi cmake cray-petsc esmf zlib craype-hugepages2M darshan
module load craype PrgEnv-intel cray-mpich
module rm craype-haswell
module load craype-mic-knl
module swap cray-mpich impi/2019.up3
module load PrgEnv-intel/6.0.5
module rm intel
module load intel/18.0.1.163
module swap craype craype/2.5.18
module rm pmi
module load pmi/5.0.14
module rm craype-haswell
module load craype-mic-knl
module rm cray-netcdf-hdf5parallel
module load cray-netcdf-hdf5parallel/4.6.1.3 cray-hdf5-parallel/1.10.2.0 cray-parallel-netcdf/1.11.1.1
module rm cmake
module load cmake/3.14.4

git clone https://github.com/E3SM-Project/scorpio.git
cd scorpio
mkdir build
cd build

CC=mpiicc CXX=mpiicpc FC=mpiifort cmake \
-DCMAKE_SYSTEM_NAME=Catamount \
-DNetCDF_PATH=/opt/cray/pe/netcdf-hdf5parallel/4.6.1.3/INTEL/16.0 \
-DPnetCDF_PATH=/opt/cray/pe/parallel-netcdf/1.11.1.1/INTEL/19.0 \
-DPIO_USE_MALLOC=ON \
..

make

[100%] Linking CXX executable spio_finfo.exe
/usr/bin/ld: /opt/cray/pe/netcdf-hdf5parallel/4.6.1.3/INTEL/16.0/lib/libnetcdf.a(libnetcdf4_la-nc4file.o): in function `att_read_var_callbk':
nc4file.c:(.text+0x71): undefined reference to `H5Aopen'
```
One possible fix seems to build scorpio with static HDF5 lib, by seting PREFER_STATIC to TRUE and adding HDF5_PATH:
```
CC=mpiicc CXX=mpiicpc FC=mpiifort cmake \
-DCMAKE_SYSTEM_NAME=Catamount \
-DPREFER_STATIC=TRUE \
-DHDF5_PATH=$HDF5_ROOT \
-DNetCDF_PATH=/opt/cray/pe/netcdf-hdf5parallel/4.6.1.3/INTEL/16.0 \
-DPnetCDF_PATH=/opt/cray/pe/parallel-netcdf/1.11.1.1/INTEL/19.0 \
-DPIO_USE_MALLOC=ON \
..
```

--- a/cime_config/machines/config_compilers.xml
+++ b/cime_config/machines/config_compilers.xml
@@ -622,7 +622,7 @@ flags should be captured within MPAS CMake files.
 
 <compiler OS="CNL">
   <CMAKE_OPTS>
-    <base> -DCMAKE_SYSTEM_NAME=Catamount</base>
+    <append MPILIB="!impi"> -DCMAKE_SYSTEM_NAME=Catamount </append>
   </CMAKE_OPTS>
   <CPPDEFS>
     <append> -DLINUX </append>

